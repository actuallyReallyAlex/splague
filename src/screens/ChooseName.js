import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Box, Button, FormField, TextInput } from 'grommet'
import {
  chooseName,
  changeScreen,
  transitionScreen
} from '../redux/actions/actions'

export class ChooseName extends Component {
  static propTypes = {
    // From mapStateToProps()
    cure: PropTypes.object.isRequired,
    // From connect()
    dispatch: PropTypes.func.isRequired,
    // From mapStateToProps()
    log: PropTypes.array.isRequired,
    // From mapStateToProps()
    plague: PropTypes.object.isRequired,
    // From mapStateToProps()
    player: PropTypes.object.isRequired,
    // From mapStateToProps()
    ui: PropTypes.object.isRequired,
    // From mapStateToProps()
    world: PropTypes.object.isRequired
  }

  state = {
    error: false,
    name: ''
  }

  playerName = React.createRef()

  handleNameSelection = e => {
    e.preventDefault()
    const { error } = this.state

    if (!error) {
      const { dispatch } = this.props
      const { name } = this.state
      dispatch(chooseName(name))
      dispatch(transitionScreen(true, 'animated fadeOut'))

      setTimeout(() => {
        dispatch(transitionScreen(false, 'animated fadeIn'))
        dispatch(changeScreen('chooseType'))
      }, 1500)
    }
  }

  handleNameChange = e => {
    e.preventDefault()
    const name = e.target.value
    const regex = /\w/gm
    if (regex.test(name)) {
      this.setState(() => ({ name, error: false }))
    } else {
      this.setState(() => ({ error: true }))
    }
  }

  render() {
    const { error, name } = this.state
    const { ui, player } = this.props
    return (
      <Box
        align="center"
        background={player.morality === 'good' ? 'accent-1' : '#252839'}
        className={ui.transitionClasses}
        fill
        justify="center"
      >
        <form>
          <Box align="center">
            <Box>
              <FormField
                error={error && "Name can't be blank. Please try again."}
                label="Name"
              >
                <TextInput
                  onChange={this.handleNameChange}
                  ref={this.playerName}
                />
              </FormField>
            </Box>
            {name && (
              <Box
                className="animated fadeInUp"
                margin={{ top: '130px' }}
                style={{ position: 'absolute' }}
              >
                <Button
                  label="Continue"
                  onClick={this.handleNameSelection}
                  primary
                  type="button"
                />
              </Box>
            )}
          </Box>
        </form>
      </Box>
    )
  }
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(ChooseName)
