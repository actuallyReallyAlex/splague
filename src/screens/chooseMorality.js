import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Box, RadioButton } from 'grommet'
import { chooseMorality } from '../redux/actions/actions'

class ChooseMorality extends Component {
  handleMoralitySelection = e => {
    const { dispatch } = this.props
    dispatch(chooseMorality(e.target.name))
  }

  render() {
    const { player } = this.props
    return (
      <Box align="center" fill justify="center">
        <Box direction="row" gap="medium">
          <RadioButton
            checked={player.morality === 'good'}
            label="Good"
            name="good"
            onChange={this.handleMoralitySelection}
          />
          <RadioButton
            checked={player.morality === 'evil'}
            label="Evil"
            name="evil"
            onChange={this.handleMoralitySelection}
          />
        </Box>
      </Box>
    )
  }
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(ChooseMorality)
