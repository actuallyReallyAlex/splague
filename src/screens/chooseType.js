import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Box, Button, FormField, Select } from 'grommet'
import { chooseType, changeScreen } from '../redux/actions/actions'
import { plagueTypes, factionTypes } from '../constants'

class ChooseType extends Component {
  state = {
    type: ''
  }

  handleBegin = () => {
    const { dispatch } = this.props
    const { type } = this.state
    dispatch(chooseType(type))
    dispatch(changeScreen('home'))
  }

  handleTypeChange = e => {
    const type = e.value
    this.setState(() => ({ type }))
  }

  render() {
    const { morality } = this.props.player
    const { type } = this.state

    return (
      <Box align="center" fill justify="center">
        <Box>
          <FormField label={morality === 'good' ? 'Faction' : 'Plague Type'}>
            <Select onChange={this.handleTypeChange} options={morality === 'good' ? factionTypes : plagueTypes} value={type} />
          </FormField>
        </Box>
        {type && (
          <Box margin={{ top: 'xlarge' }} style={{ position: 'absolute' }}>
            <Button label="Begin ..." onClick={this.handleBegin} primary />
          </Box>
        )}
      </Box>
    )
  }
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(ChooseType)
