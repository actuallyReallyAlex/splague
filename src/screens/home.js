import React from 'react'
import { connect } from 'react-redux'
import { Box } from 'grommet'
import Sidebar from '../components/Sidebar'

const Home = ({ player, ui }) => {
  return (
    <Box fill>
      <Sidebar />
    </Box>
  )
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(Home)
