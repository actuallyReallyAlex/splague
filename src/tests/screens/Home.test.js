import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import 'jest-styled-components'
import { Home } from '../../screens/Home'
import developmentState from '../fixtures/developmentState'

describe('<Home />', () => {
  it('Should render snapshot of Home screen.', () => {
    const { player, ui } = developmentState
    const renderer = new ShallowRenderer()
    renderer.render(<Home player={player} ui={ui} />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
