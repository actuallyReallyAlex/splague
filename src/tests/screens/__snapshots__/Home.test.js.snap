// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Should render snapshot of Home screen. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  background: #252839;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  width: 250px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #1b1c1d;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  padding: 12px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  height: 192px;
  width: 192px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #7D4CDB;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 6px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  background: #999999;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 6px;
  padding-right: 6px;
  padding-top: 2px;
  padding-bottom: 2px;
  border-radius: 3px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  width: 300px;
  background: #424344;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  padding: 12px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c14 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #6FFFB0;
  border-radius: 18px;
  color: #f8f8f8;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c14:hover {
  box-shadow: 0px 0px 0px 2px #6FFFB0;
}

.c6 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
}

.c4 {
  font-size: 14px;
  line-height: 20px;
}

.c8 {
  font-size: 14px;
  line-height: 20px;
  font-weight: bold;
}

.c10 {
  font-size: 12px;
  line-height: 18px;
  color: #EDEDED;
}

.c12 {
  font-size: 12px;
  line-height: 18px;
  color: #FFFFFF;
  font-weight: bold;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    margin-bottom: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    padding-left: 6px;
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    padding-top: 2px;
    padding-bottom: 2px;
  }
}

@media only screen and (max-width:768px) {
  .c13 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c13 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    font-size: 22px;
    line-height: 28px;
    max-width: 528px;
  }
}

<div
  className="animated fadeIn slow c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <span
          className="c4"
          size="small"
        >
          Image Placeholder
        </span>
      </div>
      <div
        className="c5"
      >
        <h2
          className="c6"
        />
      </div>
      <div
        className="c7"
      >
        <span
          className="c8"
          size="small"
        >
          World Information
        </span>
        <div
          className="c9"
        >
          <span
            className="c10"
            size="xsmall"
          >
            Alive Population
          </span>
          <div
            className="c11"
          >
            <span
              className="c12"
              size="xsmall"
              style={
                Object {
                  "letterSpacing": "-1px",
                }
              }
            >
              450,000,000
            </span>
          </div>
        </div>
        <div
          className="c9"
        >
          <span
            className="c10"
            size="xsmall"
          >
            Healthy Population
          </span>
          <div
            className="c11"
          >
            <span
              className="c12"
              size="xsmall"
              style={
                Object {
                  "letterSpacing": "-1px",
                }
              }
            >
              450,000,000
            </span>
          </div>
        </div>
        <div
          className="c9"
        >
          <span
            className="c10"
            size="xsmall"
          >
            Infected Population
          </span>
          <div
            className="c11"
          >
            <span
              className="c12"
              size="xsmall"
              style={
                Object {
                  "letterSpacing": "-1px",
                }
              }
            >
              0
            </span>
          </div>
        </div>
        <div
          className="c9"
        >
          <span
            className="c10"
            size="xsmall"
          >
            Dead Population
          </span>
          <div
            className="c11"
          >
            <span
              className="c12"
              size="xsmall"
              style={
                Object {
                  "letterSpacing": "-1px",
                }
              }
            >
              0
            </span>
          </div>
        </div>
      </div>
      <div
        className="c7"
      >
        <span
          className="c8"
          size="small"
        >
          Player Information
        </span>
        <div
          className="c9"
        >
          <span
            className="c10"
            size="xsmall"
          >
            Class Type
          </span>
          <div
            className="c11"
          >
            <span
              className="c12"
              size="xsmall"
              style={
                Object {
                  "letterSpacing": "-1px",
                }
              }
            />
          </div>
        </div>
        <div
          className="c9"
        >
          <span
            className="c10"
            size="xsmall"
          >
            Plague Mutations
          </span>
          <div
            className="c11"
          >
            <span
              className="c12"
              size="xsmall"
              style={
                Object {
                  "letterSpacing": "-1px",
                }
              }
            >
              0
            </span>
          </div>
        </div>
      </div>
      <div
        className="c7"
      >
        <span
          className="c8"
          size="small"
        >
          Cure Information
        </span>
        <div
          className="c9"
        >
          <span
            className="c10"
            size="xsmall"
          >
            Completion
          </span>
          <div
            className="c11"
          >
            <span
              className="c12"
              size="xsmall"
              style={
                Object {
                  "letterSpacing": "-1px",
                }
              }
            >
              0 %
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="c13"
    >
      <button
        className="c14"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="button"
      >
        <div
          className="c15"
        >
          Test
        </div>
      </button>
    </div>
  </div>
</div>
`;
